sstables/sstable_set.cc

```diff
@@ -946,9 +946,9 @@ compound_sstable_set::compound_sstable_set(schema_ptr schema, std::vector<lw_sha
 std::unique_ptr<sstable_set_impl> compound_sstable_set::clone() const {
     std::vector<lw_shared_ptr<sstable_set>> cloned_sets;
     cloned_sets.reserve(_sets.size());
-    for (auto& set : _sets) {
+    for (const auto& set : _sets) {
         // implicit clone by using sstable_set's copy ctor.
-        cloned_sets.push_back(make_lw_shared(std::move(*set)));
+        cloned_sets.push_back(make_lw_shared(*set));
```

Comment:
what's wrong with `cloned_sets.push_back(set)`?
we don't need to explicitly call `make_lw_shared(*set)`, it's confusing IMO.

Reply:
I see. Then to make this clearer I suggest to clone the set in a separate statement and do something like:
```c++
    auto cloned_set = *set;
    cloned_sets.push_back(cloned_set.shared_from_this());
```

Reply:
What was wrong with the previous code `make_lw_shared(*set)`? I don't think it gets clearer than that. The suggested code leads to use-after-free and illegal free.